src/buffer.h

```diff
@@ -150,10 +150,29 @@ class Buffer {
   /// initial count. This requires the format to have been set.
   void ResizeTo(uint32_t element_count);
 
+  /// Resizes the buffer to hold |size_in_bytes| over format_->SizeInBytes()
+  /// number of elements while resizing the buffer to |size_in_bytes| bytes.
+  /// This requires the format to have been set. This is separate from
+  /// ResizeTo() since the given argument here is |size_in_bytes| bytes vs
+  /// |element_count| elements
+  void SetSizeInBytes(uint32_t size_in_bytes);
+
+  /// Sets the max_size_in_bytes_ to |max_size_in_bytes| bytes
+  void SetMaxSizeInBytes(uint32_t max_size_in_bytes);
+  /// Returns max_size_in_bytes_ if it is not zero. Otherwise it means this
+  /// buffer is an amber buffer which has a fix size and returns
+  /// GetSizeInBytes()
+  uint32_t GetMaxSizeInBytes() const;
+
   /// Write |data| into the buffer |offset| bytes from the start. Write
   /// |size_in_bytes| of data.
   Result SetDataWithOffset(const std::vector<Value>& data, uint32_t offset);
 
+  /// At each ubo, ssbo size and ssbo subdata size calls, recalculates
```

Comment:
All together, these size-related changes make a *lot* more sense to me.  The new method name here is a big improvement.